index.rst

```diff
@@ -50,7 +50,696 @@
++---------------------------+--------+----------------------------------------+
+| region                    | blob   |                                        |
++---------------------------+--------+----------------------------------------+
+| UNIQUE (index, tract_id)                                                    |
++---------------------------+--------+----------------------------------------+
+
+Tracts are subdivided into Patches, which share the Tract coordinate system
+and define similarly-sized regions that overlap by a configurable amount.  As
+with Tracts, we may want to include fields to describe Patch boundaries in this
+table in the future.
+
+.. _cs_calibration_dataunits:
+
+Calibration DataUnits
+---------------------
+
++---------------------------+--------+-------------------------------------------------+
+| *CalibRange*                                                                         |
++===========================+========+=================================================+
+| calib_range_id            | uint64 | PRIMARY KEY                                     |
++---------------------------+--------+-------------------------------------------------+
+| first_visit               | uint64 | NOT NULL                                        |
++---------------------------+--------+-------------------------------------------------+
+| last_visit                | uint64 |                                                 |
++---------------------------+--------+-------------------------------------------------+
+| camera_id                 | uint64 | NOT NULL REFERENCES Camera (camera_id)          |
++---------------------------+--------+-------------------------------------------------+
+| physical_filter_id        | uint64 | REFERENCES PhysicalFilter (physical_filter_id)  |
++---------------------------+--------+-------------------------------------------------+
+| UNIQUE (first_visit, last_visit, camera_id, physical_filter_id)                      |
++---------------------------+--------+-------------------------------------------------+
+
++------------------------+--------+-----------------------------------------------------------+
+| *SensorCalibRange*                                                                          |
++========================+========+===========================================================+
+| sensor_calib_range_id  | uint64 | PRIMARY KEY                                               |
++------------------------+--------+-----------------------------------------------------------+
+| first_visit            | uint64 | NOT NULL                                                  |
++------------------------+--------+-----------------------------------------------------------+
+| last_visit             | uint64 |                                                           |
++------------------------+--------+-----------------------------------------------------------+
```

Comment:
I'm not entirely sure this isn't too restrictive. This requires that a range is always contiguous and applies to all visits taken in the range. Why not instead have a many-to-many join table?

Reply:
Since first_visit and last_visit can be the same thing, it's no more restrictive.  The question is whether it's efficient, and I'm not sure.  This is the approach (albeit with datetime ranges instead of visit ranges) that we use now, but I don't know how big the ranges end up being.

I could certainly imagine it being useful to make the join table the source of truth (and hence force it to be materialized rather than a view).